### PR TITLE
Add S3 async client and s3 utils

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/aws/utils/S3Utils.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import monix.catnap.syntax._
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.services.s3.model.{GetObjectRequest, PutObjectRequest, PutObjectResponse}
+
+class S3Utils(client: S3AsyncClient) {
+  def upload(bucket: String, key: String, path: Path): IO[PutObjectResponse] = {
+    IO(client.putObject(PutObjectRequest.builder.bucket(bucket).key(key).build, path)).futureLift
+  }
+
+  def downloadFiles(bucket: String, key: String, path: Option[Path] = None) = {
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/aws/utils/S3Utils.scala]: Maybe explicitly state the return type for this function for consistency: `IO[GetObjectResponse]`

**Review by TomJKing — COMMENTED** [review]
Looks good. One very minor suggestion. Also is it possible to right some unit tests for the `S3Utils`?

File: src/main/scala/uk/gov/nationalarchives/aws/utils/S3Utils.scala (hunk 1 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/aws/utils/S3Utils.scala]: I've added this in. I've put some tests in for the s3 utils as well.

File: src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    val s3Utils = new S3Utils(s3AsyncClient)
+    s3Utils.downloadFiles("bucket", "key", Path.of("path").some).unsafeRunSync()
+    val request: GetObjectRequest = requestCaptor.getValue
+
+    request.bucket() should equal("bucket")
+    request.key() should equal("key")
+    pathCaptor.getValue.getFileName.toString should equal("path")
+  }
+
+  "The download method" should "use the key for the download path if a key is not provided" in {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala]: Should this test read: "use the key for the download path if a **download path** is not provided"

**Review by TomJKing — COMMENTED** [review]
Just one more minor comment

File: src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala (hunk 2 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala]: It should indeed, good spot, cheers.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**